### PR TITLE
fix #113 cmd palette

### LINKED ISSUE #113: Allow for dev mode commands to be accessed from command palette

### PR DESCRIPTION
**Pull request by travis1111**
Added liberty dev ex commands to vscode command palette.

### CODE REVIEW

File: src/util/helperUtil.ts [hunk 1]
```diff
@@ -20,3 +22,20 @@ export function getConfiguration<T>(section: string, resourceOrFilepath?: vscode
...
+/**
+ * Filters the projects by command.
+ * @param projects The list of projects
+ * @param commamnd The command to check
+ * @returns a list of projects that the given command can be excuted on.
+ */
+export function filterProjects(projects: LibertyProject[], commamnd: string): LibertyProject[] {
+	let resultProjects: LibertyProject[] = [];
+	for ( const project of projects) {
+		const applicableTypes = COMMAND_AND_PROJECT_TYPE_MAP["liberty.dev.start"];
```
Comment (kathrynkodama) [inline src/util/helperUtil.ts:35]: Should "liberty.dev.start" here be replaced with the command?

File: src/util/helperUtil.ts [hunk 2]
```diff
@@ -20,3 +22,20 @@ export function getConfiguration<T>(section: string, resourceOrFilepath?: vscode
...
 	return vscode.workspace.getConfiguration("liberty", resource).get<T>(section);
 }
+
+/**
+ * Filters the projects by command.
+ * @param projects The list of projects
+ * @param commamnd The command to check
+ * @returns a list of projects that the given command can be excuted on.
+ */
+export function filterProjects(projects: LibertyProject[], commamnd: string): LibertyProject[] {
```
Comment (kathrynkodama) [inline src/util/helperUtil.ts:32]: typo in command

File: package.json [hunk 3]
```diff
@@ -74,7 +74,7 @@
 			{
 				"command": "liberty.dev.custom",
 				"category": "Liberty",
-				"title": "Start..."
+				"title": "Start custom"
```
[the remark this answers is not in the document]
Comment (kathrynkodama) [reply package.json:77]: Will that interfere with "..." naming in the command palette?

[the remark this answers is not in the document]
Comment (travis1111) [reply package.json:77]: No.

File: src/liberty/libertyProject.ts [hunk 4]
```diff
@@ -25,11 +27,23 @@ export class ProjectProvider implements vscode.TreeDataProvider<LibertyProject>
...
+
+	public static setInstance(projectProvider: ProjectProvider) {
+		ProjectProvider.instance = projectProvider;
+	}
+	public getProjects() {
+		return this.projects;
+	}
 	public async refresh(): Promise<void> {
 		// update the map of projects
+		var statusMessage = vscode.window.setStatusBarMessage("Refreshing Liberty Dev Dashboard projects...");
```
[the remark this answers is not in the document]
Comment (travis1111) [reply src/liberty/libertyProject.ts]: addressed

File: src/util/helperUtil.ts (hunk 1 above)
Comment (travis1111) [reply src/util/helperUtil.ts]: addressed

File: src/util/helperUtil.ts (hunk 2 above)
Comment (travis1111) [reply src/util/helperUtil.ts]: addressed

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (TrevCraw) [reply package.json:77]: Suggestion to just pick one for now and then open an issue to raise to the design team.

[the remark this answers is not in the document]
Comment (kathrynkodama) [reply package.json:77]: Sounds good, @travis1111 I think going back to Start... for now makes sense

**Review by kathrynkodama — APPROVED** [review]

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (travis1111) [reply package.json]: Its back to `Start...` now

File: src/util/helperUtil.ts [hunk 5]
```diff
@@ -20,3 +22,20 @@ export function getConfiguration<T>(section: string, resourceOrFilepath?: vscode
 	}
 	return vscode.workspace.getConfiguration("liberty", resource).get<T>(section);
 }
+
+/**
+ * Filters the projects by command.
+ * @param projects The list of projects
+ * @param commamnd The command to check
```
Comment (TrevCraw) [inline src/util/helperUtil.ts:29]:
```suggestion
 * @param command The command to check
```

**Review by TrevCraw — APPROVED** [review]

**Pull request merged**